otter/cloud_client.py

```diff
@@ -552,6 +555,48 @@ def _parse_known_errors(code, json_body):
     return eff.on(error=_parse_known_errors)
 
 
+def remove_clb_nodes(lb_id, node_ids):
+    """
+    Remove multiple nodes from a load balancer.
+
+    :param str lb_id: A load balancer ID.
+    :param node_ids: iterable of node IDs.
+    :return: Effect of None.
+
+    Succeeds on 202.
+
+    This function will handle the case where *some* of the nodes are valid and
+    some aren't, by retrying deleting only the valid ones.
+    """
+    node_ids = map(str, node_ids)
+    eff = service_request(
+        ServiceType.CLOUD_LOAD_BALANCERS,
+        'DELETE',
+        append_segments('loadbalancers', lb_id, 'nodes'),
+        params={'id': node_ids},
+        success_pred=has_code(202))
+
+    def check_invalid_nodes(exc_info):
+        code = exc_info[1].code
+        body = exc_info[1].body
+        if code == 400:
+            message = try_json_with_keys(
+                body, ["validationErrors", "messages", 0])
+            match = _CLB_NODE_REMOVED_PATTERN.match(message)
+            if match:
+                removed = concat([group.split(',')
+                                  for group in match.groups()])
+                return remove_clb_nodes(lb_id, set(node_ids) - set(removed))
```

Comment:
Actually apologies, maybe the decision is blocking, though, because there is a duplicate test in both steps and cloud_client, and we should probably decide where it should go :)